marklogic-client-api/src/main/java/com/marklogic/client/impl/HTTPSamlAuthInterceptor.java

```diff
@@ -74,26 +72,22 @@ public Response intercept(Chain chain) throws IOException {
         return chain.proceed(authenticatedRequest);
     }
 
-    private void authorizeCallback() {
-        try {
-            expiringSAMLAuth = authorizer.apply(expiringSAMLAuth);
-
-            if (expiringSAMLAuth == null) {
-                throw new IllegalArgumentException("SAML Authentication cannot be null");
-            }
+    private synchronized void authorizeCallback() {
```

Comment:
To eliminate the possibility of a race condition, please restore the old expiry parameter and the testing of the old expiry from the version of this method prior to the isCallbackExecuting revision.